pkg/api/v1/schema.go

```diff
@@ -32,6 +32,15 @@ import (
  ATTENTION: Rerun code generators when comments on structs or fields are modified.
 */
 
+// ConfigMapSource represents a reference to a ConfigMap in the same namespace.
+// More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/
+// ---
+// +k8s:openapi-gen=true
+type ConfigMapSource struct {
```

Comment:
The struct should look like this (taken and adapted from k8s ConfigMapVolumeSource):

```go
type ConfigMapVolumeSource struct {
	LocalObjectReference `json:",inline" protobuf:"bytes,1,opt,name=localObjectReference"`
	// If unspecified, each key-value pair in the Data field of the referenced
	// ConfigMap will be projected into the volume as a file whose name is the
	// key and content is the value. If specified, the listed keys will be
	// projected into the specified paths, and unlisted keys will not be
	// present. If a key is specified which is not present in the ConfigMap,
	// the volume setup will error unless it is marked optional. Paths must be
	// relative and may not contain the '..' path or start with '..'.
	// +optional
	Items []KeyToPath `json:"items,omitempty" protobuf:"bytes,2,rep,name=items"`
	// Optional: mode bits to use on created files by default. Must be a
	// value between 0 and 0777. Defaults to 0644.
	// Directories within the path are not affected by this setting.
	// This might be in conflict with other options that affect the file
	// mode, like fsGroup, and the result can be other mode bits set.
	// +optional
	Optional *bool `json:"optional,omitempty" protobuf:"varint,4,opt,name=optional"`
}
```

Reply:
According to your comment, I changed structures, however, if I will specify items for ConfigMap or Secret then I receive failure from ```kubectl```, ```kubectl``` validation is sending me an error:
```
validation failure list
type in spec.volumes.configMap is required
```
I even checked the above example for ```disks```, like changing ```disks``` name to ```items```, the same result.

I don't know why I cannot set ```items``` name in a list in YAML file, I believe it might be related to "registered resources" (?), I have to investigate that, my current knowledge about this is too small now to say something more...

Reply:
It seems it's related to swagger and reserved ```items``` keyword.

Reply:
I think this is a bug in k8s CRD validation, I just made simple experiments like:
* changing ```items``` to ```item```, it works
* removing ```configMap``` validation from ```manifests/generated/vmi-resource.yaml``` and applying file again, works as well

After all, you cannot use ```items``` name as a parameter name.